### PR TITLE
fix: persist build time spi options

### LINKED ISSUE #33902: Not persisted config settings prevent server start
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

dist/quarkus

### Describe the bug

Starting a newly built Keycloak 26.0.0 with ``kc start --optimized`` shows the following error and starting fails.

> ERROR: The following build time options have values that differ from what is persisted - the new values will NOT be used until another build is run: kc.spi-user-cache-default-enabled

When running ``kc build`` with have the following config setting in *keycloak.conf*:
``spi-user-cache-default-enabled=false``

Obviously the config parameter ``spi-user-cache-default-enabled`` is not added to the */generated-bytecode.jar/META-INF/keycloak-persisted.properties*, but at startup the ``org.keycloak.quarkus.runtime.cli.Picocli.checkSpiOptions(IncludeOptions, List<String>, List<String>)`` method checks current settings (including those from *keycloak.conf*) against the entries of this file. The ``currentValueStr=false`` does not match ``currentValue=null`` from the persisted properties, hence the error.

In Keyckoak 25.0.6 this problem did not exist! There was no ``checkSpiOptions()`` in the **Picocli** class.

### Version

26.0.0

### Regression

- [X] The issue is a regression

### Expected behavior

All properties set at build-time must be recognized correctly at startup-time.

### Actual behavior

Starting Keycloak with --optimized fails when ``spi-user-cache-default-enabled=false`` has been set at build-time.

### How to Reproduce?

1. Put ``spi-user-cache-default-enabled=false`` into /conf/keycloak.conf
2. Run kc build
3. Run kc start --optimized

### Anything else?

This is not just related to property ``spi-user-cache-default-enabled``. This error will also occur with properties matching one of these patterns:
- kc.spi*-provider
- kc.spi*-provider-default
- kc.spi*-enabled

### PR DESCRIPTION
**Pull request by shawkins**
closes: #33902

This should test cleanly after #34156 and a test tweak is incorporated.

~~There will be a later follow-up to enable the arquillian tests to just use auto-builds.~~ - auto-builds seem to be sufficient.

### CODE REVIEW

**Comment by mabartos** [comment]
@shawkins What's the status of this PR?

**Comment by shawkins** [comment]
> @shawkins What's the status of this PR?

Needs the other pr merged first.

**Comment by shawkins** [comment]
Ready now.

File: testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/client/KeycloakTestingClient.java [hunk 1]
```diff
@@ -81,9 +81,9 @@ public TestingResource testing(String realm) {
     }
 
     public void enableFeature(Profile.Feature feature) {
-        Set<Profile.Feature> enabledFeatures = testing().enableFeature(feature.toString());
-        Assert.assertFalse(enabledFeatures.contains(feature));
-        ProfileAssume.updateDisabledFeatures(enabledFeatures);
+        Set<Profile.Feature> disabledFeatures = testing().enableFeature(feature.toString());
+        Assert.assertFalse(disabledFeatures.contains(feature));
+        ProfileAssume.updateDisabledFeatures(disabledFeatures);
```
Comment (vmuzikar) [inline testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/client/KeycloakTestingClient.java:86]: Is this intentional?

**Review by vmuzikar — APPROVED** [review]
Overall LGTM.

File: testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/client/KeycloakTestingClient.java (hunk 1 above)
Comment (shawkins) [reply testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/client/KeycloakTestingClient.java:86]: yes, that method always returns the disabled features, so it was misnamed.

**Comment by vmuzikar** [comment]
@shawkins Can you please create the backport?

**Comment by shawkins** [comment]
> @shawkins Can you please create the backport?

@vmuzikar https://github.com/keycloak/keycloak/pull/34261

**Pull request merged**